test/OpenIddict.Server.Tests/OpenIddictServerBuilderTests.cs

```diff
@@ -0,0 +1,2000 @@
+        {
+            // Arrange
+            var services = CreateServices();
+            var builder = CreateBuilder(services);
+
+            // Act
+            builder.AddEventHandler<CustomContext>(x =>
+            {
+                x.UseSingletonHandler(new CustomHandler());
+            });
+
+            // Assert
+            Assert.Contains(services, service =>
+                service.ServiceType == typeof(CustomHandler) &&
+                service.ImplementationInstance?.GetType() == typeof(CustomHandler) &&
+                service.Lifetime == ServiceLifetime.Singleton);
+        }
+
+        [Fact]
+        public void AddEventHandler_SingletonHandlerIsRegisteredAsASingleton()
+        {
+            // Arrange
+            var services = CreateServices();
+            var builder = CreateBuilder(services);
+
+            // Act
+            builder.AddEventHandler<CustomContext>(x =>
+            {
+                x.UseSingletonHandler<CustomHandler>();
+            });
+
+            // Assert
+            Assert.Contains(services, service =>
+                service.ServiceType == typeof(CustomHandler) &&
+                service.Lifetime == ServiceLifetime.Singleton);
+        }
+
+        [Fact]
+        public void AddEventHandler_ScopedHandlerIsRegisteredAsScoped()
+        {
+            // Arrange
+            var services = CreateServices();
+            var builder = CreateBuilder(services);
+
+            // Act
+            builder.AddEventHandler<CustomContext>(x =>
+            {
+                x.UseScopedHandler<CustomHandler>();
+            });
+
+            // Assert
+            Assert.Contains(services, service =>
+                service.ServiceType == typeof(CustomHandler) &&
+                service.Lifetime == ServiceLifetime.Scoped);
+        }
+
+        [Fact]
+        public void AddEncryptionCredentials_ThrowsExceptionWhenCredentialsAreNull()
+        {
+            // Arrange
+            var services = CreateServices();
+            var builder = CreateBuilder(services);
+            EncryptingCredentials credentials = null;
+
+            // Act
+            Action action = () => builder.AddEncryptionCredentials(credentials);
+
+            // Assert
+            var exception = Assert.Throws<ArgumentNullException>(action);
+            Assert.Equal(nameof(credentials), exception.ParamName);
+        }
+
+        [Fact]
+        public void AddEncryptionKey_ThrowsExceptionWhenKeyIsNull()
+        {
+            // Arrange
+            var services = CreateServices();
+            var builder = CreateBuilder(services);
+            SecurityKey key = null;
+
+            // Act
+            Action action = () => builder.AddEncryptionKey(key);
+
+            // Assert
+            var exception = Assert.Throws<ArgumentNullException>(action);
+            Assert.Equal(nameof(key), exception.ParamName);
+        }
+
+        private class FakeAsymmetricSecurityKey : AsymmetricSecurityKey
```

Comment:
May I suggest using Moq here? I used in the ASOS tests and it worked fine in this case: https://github.com/aspnet-contrib/AspNet.Security.OpenIdConnect.Server/blob/dev/test/AspNet.Security.OpenIdConnect.Server.Tests/OpenIdConnectServerExtensionsTests.cs#L530

Reply:
Verified tests still pass using Moq, so replaced with Moq.